### PR TITLE
add debug APIs and update GPU temp memory size

### LINKED ISSUE #1976: Copy index data from cpu to gpu memory get slower with nq increase
**Describe the bug**
Copy index data from cpu to gpu memory get slower with nq increase

**Steps/Code to reproduce behavior**
1. set gpu_search_threshold to 1 in server_config.yaml
2. start milvus(gpu edition)
3. create a collection, index_file_size = 1024MB, dimension = 1024
4. insert 1,000,000 vectors
5. create index sq8, nlist=16384
6. restart milvus server, preload the table, search, nq=**2**, topk=10, nprobe=16
result: from the log each index file load from cpu to gpu cost about **50ms**
7. restart milvus server, preload the table, search, nq=**200**, topk=10, nprobe=16
result: from the log each index file load from cpu to gpu cost about **300ms**
8. restart milvus server, preload the table, search, nq=**1000**, topk=10, nprobe=16
result: from the log each index file load from cpu to gpu cost about **1000ms**

### PR DESCRIPTION
**Pull request by cydrain**
Signed-off-by: yudong.cai <yudong.cai@zilliz.com>

**What type of PR is this?**
feature

**What this PR does / why we need it:**
add some debug APIs and update GPU temp memory size to 256M

**Which issue(s) this PR fixes:**
Fixes #1976

**Special notes for your reviewer:**

**Does this PR introduce a user-facing change?:**

**Additional documentation (e.g. design docs, usage docs, etc.):**

### CODE REVIEW

File: core/src/scheduler/tasklabel/SpecResLabel.h [hunk 1]
```diff
@@ -35,14 +35,13 @@ class SpecResLabel : public TaskLabel {
         return resource_;
     }
 
-    inline std::string&
-    resource_name() {
-        return resource_name_;
+    virtual inline std::string
```
Comment (JinHai-CN) [inline core/src/scheduler/tasklabel/SpecResLabel.h:38]: override

File: core/src/scheduler/tasklabel/TaskLabel.h [hunk 2]
```diff
@@ -28,6 +29,11 @@ class TaskLabel {
         return type_;
     }
 
+    virtual inline std::string
```
Comment (JinHai-CN) [inline core/src/scheduler/tasklabel/TaskLabel.h:32]: inline std::string name() const { ...

File: core/src/index/archive/KnowhereResource.cpp [hunk 3]
```diff
@@ -70,8 +70,8 @@ KnowhereResource::Initialize() {
         return Status::OK();
 
     struct GpuResourceSetting {
-        int64_t pinned_memory = 300 * M_BYTE;
-        int64_t temp_memory = 300 * M_BYTE;
+        int64_t pinned_memory = 256 * M_BYTE;
```
Comment (scsven) [inline core/src/index/archive/KnowhereResource.cpp:73]: This may cause performance degradation

File: core/src/scheduler/resource/Resource.cpp [hunk 4]
```diff
@@ -153,7 +153,7 @@ Resource::pick_task_execute() {
 
 void
 Resource::loader_function() {
-    SetThreadName("taskloader_th");
+    SetThreadName("task_loader_th");
```
Comment (scsven) [inline core/src/scheduler/resource/Resource.cpp:156]: TOO LONG. The thread name is a meaningful C language string, whose length is restricted to 16 characters, including the terminating null byte ('\0').
http://man7.org/linux/man-pages/man3/pthread_setname_np.3.html

File: core/src/scheduler/resource/Resource.cpp [hunk 5]
```diff
@@ -184,7 +184,7 @@ Resource::loader_function() {
 
 void
 Resource::executor_function() {
-    SetThreadName("taskexector_th");
+    SetThreadName("task_executor_th");
```
Comment (scsven) [inline core/src/scheduler/resource/Resource.cpp:187]: TOO LONG.

**Review by wxyucs — CHANGES_REQUESTED** [review]

File: core/src/scheduler/resource/Resource.cpp (hunk 5 above)
Comment (cydrain) [reply core/src/scheduler/resource/Resource.cpp]: use "taskexecutor_th" now, it's 15 charactors

File: core/src/scheduler/resource/Resource.cpp (hunk 4 above)
Comment (cydrain) [reply core/src/scheduler/resource/Resource.cpp]: rollback

File: core/src/index/archive/KnowhereResource.cpp (hunk 3 above)
Comment (cydrain) [reply core/src/index/archive/KnowhereResource.cpp:73]: we'll check the nightly test result

Comment (cydrain) [reply core/src/index/archive/KnowhereResource.cpp:73]: we'll check the nightly testing result

**Review by JinHai-CN — APPROVED** [review]

**Review by shengjun1985 — APPROVED** [review]

**Review by wxyucs — APPROVED** [review]

**Review by JinHai-CN — APPROVED** [review]

**Pull request merged**